### ISSUE TITLE
Order in rtp: '~/.vim/pack/.../start/.../indent/' after '~/.vim/after/ftplugin/'

### ISSUE THREAD

**Issue by kiryph** [vim/vim#3333]
I wanted to adjust the setting `'indentkeys'` for tex files in my `~/.vim/after/ftplugin/tex.vim` file. The default value is already changed by the ftplugin `vimtex` in `indent/tex.vim`.

I was surpised that the pack path with the folder indent comes after the `'after/ftplugin'`. I added a few `echom` to figure this out:

```
~/.vim/pack/minpac/start/vimtex/ftplugin/tex.vim
~/.vim/after/ftplugin/tex.vim
~/.vim/pack/minpac/start/vimtex/indent/tex.vim
~/.vim/after/indent/tex.vim  
```
I could resolve my issue by creating the file `~/.vim/after/indent/tex.vim`.

However, I was wondering whether this is by intention or should all after directories come after pack paths?

The order would look like
```
~/.vim/pack/minpac/start/vimtex/ftplugin/tex.vim
~/.vim/pack/minpac/start/vimtex/indent/tex.vim
~/.vim/after/ftplugin/tex.vim
~/.vim/after/indent/tex.vim  
```

My vim version is 8.1.0072.
(according to ftp://ftp.vim.org/pub/vim/patches/8.1/README there was no patch up to 8.1.0288 which changes the load order after 8.1.0072)

**Comment by tonymec** [comment]
```
~/.vim/pack/minpac/start/vimtex/ftplugin/tex.vim
~/.vim/after/ftplugin/tex.vim
~/.vim/pack/minpac/start/indent/tex.vim
~/.vim/after/indent/tex.vim  
```
This is indeed expected.
The first two are filetype-plugins, sourced as soon as `'filetype'` is set to `tex`. IIUC when `:filetype plugin on` (with or without `indent`) is set, Vim defines the autocommand
```vim
au FileType * runtime! <amatch>.vim
```
(see `:help :runtime`).
The other two are indent scripts, they are sourced by virtue of the code at lines 11-31 of `$VIMRUNTIME/indent.vim`. This code defines additional autocommands which are in their own autocommand group, so that `:filetype indent off` can disable them without influencing the sourcing of filetype-plugins.

Best regards,
Tony.

**Comment by chrisbra** [comment]
Closing. Please ask user questions on the vim_use mailinglist or use vi.stackexchange.com. This here is a bug tracker for problems with Vim. We cannot provide individual help here.

**Comment by kiryph** [comment]
@tonymec Thank you very much. This was very helpful. And also thanks that you did not simply close it as a user question.

@chrisbra 
> Please ask user questions on the vim_use mailinglist or use vi.stackexchange.com.

Sorry. I will do this and actually have done before. Obviously, I was suspecting that this might have been unintentional.

A naive user like me could think that `:h after-directory` comes after everything.

Apparently, I was wrong.

I also checked the help sections `:h :filetype-indent-on`, `:h 'rtp'`, `:h 'indentkeys'`,  `:h 'indentexpr'`, `:h packages`. None of them indicates this. IMHO this could be illustrated in the help section `after-directory` and a reminder could be added to `'indentkeys'`, `'indentexpr'` and possibly `filetype-indent-on`.

I could verify the event order for myself with
```
:verbose autocmd Filetype *   
filetypeplugin  FileType
    *         call s:LoadFTPlugin()
        Last set from /usr/local/Cellar/vim/HEAD-2dfcef4/share/vim/vim81/ftplugin.vim
filetypeindent  FileType
    *         call s:LoadIndent()
        Last set from /usr/local/Cellar/vim/HEAD-2dfcef4/share/vim/vim81/indent.vim
syntaxset  FileType
    *         exe "set syntax=" . expand("<amatch>")
        Last set from /usr/local/Cellar/vim/HEAD-2dfcef4/share/vim/vim81/syntax/syntax.vim
```

I also did not know that all runtime scripts are sourced through autocmds.
The order of events according to https://vi.stackexchange.com/questions/4493/what-is-the-order-of-winenter-bufenter-bufread-syntax-filetype-events#comment6681_4495 has changed in the past and is not well-defined in the case of applying to multiple files.

My impression is that this was rather expert knowledge beyond vim `:help`. Therefore I do not think that raising this on the bug tracker was completely outrageous.

Nevertheless, sorry for the noise again. It won't happen again.

---

**UPDATE**

I have also checked the load order without the plugin just `$VIMRUNTIME` and `~/.vim/after`. No surprise anymore:

```
$VIMRUNTIME/ftplugin/tex.vim
~/.vim/after/ftplugin/tex.vim
$VIMRUNTIME/indent/tex.vim
~/.vim/after/indent/tex.vim  
$VIMRUNTIME/syntax/tex.vim
~/.vim/after/syntax/tex.vim  
```

IMHO following part in `:h 'rtp'` is in this case misleading:
```
				*'runtimepath'* *'rtp'* *vimfiles*
'runtimepath' 'rtp'	string	(default:
					Unix, Mac OS X: "$HOME/.vim,
						$VIM/vimfiles,
						$VIMRUNTIME,
						$VIM/vimfiles/after,
						$HOME/.vim/after"
```
and could be improved by mentioning that this order only applies to the same kind of runtime file.
For example, the order across different kinds of runtime files specific for filetypes, i.e. ftplugin, indent, syntax enabled with `:filetype plugin indent on` and `:syntax on` is 
```
{"$HOME/.vim, $VIM/vimfiles, $VIMRUNTIME, $VIM/vimfiles/after, $HOME/.vim/after"}/ftplugin/
{"$HOME/.vim, $VIM/vimfiles, $VIMRUNTIME, $VIM/vimfiles/after, $HOME/.vim/after"}/indent/
{"$HOME/.vim, $VIM/vimfiles, $VIMRUNTIME, $VIM/vimfiles/after, $HOME/.vim/after"}/syntax/
```

This means for example files in `$HOME/.vim/after/ftplugin/` come before all indent and syntax files.

---

**SECOND UPDATE**

The order actually depends on the order of the invocation of

```
:filetype indent on
:filetype plugin on
:syntax on
```
This means that changing a setting should always be done in the same kind of runtime file.
